pkg/cluster/fetch.go

```diff
@@ -16,19 +16,36 @@ type BundlesFetch func(ctx context.Context, name, namespace string) (*v1alpha1re
 
 type GitOpsFetch func(ctx context.Context, name, namespace string) (*v1alpha1.GitOpsConfig, error)
 
+type FluxConfigFetch func(ctx context.Context, name, namespace string) (*v1alpha1.FluxConfig, error)
+
 type EksdReleaseFetch func(ctx context.Context, name, namespace string) (*eksdv1alpha1.Release, error)
 
 type OIDCFetch func(ctx context.Context, name, namespace string) (*v1alpha1.OIDCConfig, error)
 
-func BuildSpecForCluster(ctx context.Context, cluster *v1alpha1.Cluster, bundlesFetch BundlesFetch, eksdReleaseFetch EksdReleaseFetch, gitOpsFetch GitOpsFetch, oidcFetch OIDCFetch) (*Spec, error) {
+func BuildSpecForCluster(ctx context.Context, cluster *v1alpha1.Cluster, bundlesFetch BundlesFetch, eksdReleaseFetch EksdReleaseFetch, gitOpsFetch GitOpsFetch, fluxConfigFetch FluxConfigFetch, oidcFetch OIDCFetch) (*Spec, error) {
 	bundles, err := GetBundlesForCluster(ctx, cluster, bundlesFetch)
 	if err != nil {
 		return nil, err
 	}
-	gitOpsConfig, err := GetGitOpsForCluster(ctx, cluster, gitOpsFetch)
-	if err != nil {
-		return nil, err
+
+	var fluxConfig *v1alpha1.FluxConfig
+	var gitOpsConfig *v1alpha1.GitOpsConfig
```

Comment:
all support for the GitOpsConfig will be short lived, as we will remove it entirely in a near future minor version.

while we're converting the GitOpsConfig to a FluxConfig for internal use, when we marshall it and write it to the cluster, we still write it as a  GitOpsConfig crd, so that a user who provides a GitOpsConfig won't get a confusing result like us replacing their GitOpsConfig with a flux config. So when fetching objects from the cluster for inspection we still need to check for the existence of the GitOpsConfig